src/gmp/models/report/os.js

```diff
@@ -22,6 +23,31 @@ class OperatingSystem {
     }
   }
 
+  addHostCompliance(host, compliance) {
+    if (!(host.ip in this.hosts.complianceByIp)) {
+      this.hosts.complianceByIp[host.ip] = compliance;
+    }
+    const isNoInCompliance = Object.values(this.hosts.complianceByIp).some(
```

Comment:
Nitpick: Could extract `Object.values(this.hosts.complianceByIp)` and avoid duplicates.